ReactiveUI.Fody/ReactiveUIPropertyWeaver.cs

```diff
@@ -54,6 +58,59 @@ public void Execute()
             {
                 foreach (var property in targetType.Properties.Where(x => x.IsDefined(reactiveAttribute)).ToArray())
                 {
+                    TypeReference genericTargetType = targetType;
+                    if (targetType.HasGenericParameters)
+                    {
+                        var genericDeclaration = new GenericInstanceType(targetType);
+                        foreach (var parameter in targetType.GenericParameters)
+                        {
+                            genericDeclaration.GenericArguments.Add(parameter);
+                        }
+                        genericTargetType = genericDeclaration;
+                    }
+
+                    if (property.SetMethod == null && property.GetMethod.IsExpressionBodiedProperty())
+                    {
+                        // Add this.RaisePropertyChange(PropertyName) to the setter of any properties referenced 
+                        // by this expression bodied property where they have an available setter
+                        var getInstructions = property.GetMethod.Body.Instructions.Where(i =>
+                        {
+                            var operand = i.Operand as MethodDefinition;
+                            return i.OpCode == OpCodes.Call && (operand?.IsGetter ?? false);
+                        });
+
+                        var setMethodsForGetInstructions =
+                            targetType.Properties.Where(
+                                    prop =>
+                                        prop.SetMethod != null &&
+                                        prop.SetMethod.HasBody &&
+                                        getInstructions.Any(
+                                            i => ((MethodDefinition) i.Operand).FullName == prop.GetMethod.FullName))
+                                .Select(prop => prop.SetMethod);
+
+                        if (!setMethodsForGetInstructions.Any())
+                        {
+                            LogError(
+                                $"Property {property.DeclaringType.FullName}.{property.Name} only references other expression bodied properties, as these have no setter they're not supported by the [Reactive] attribute");
+                        }
+
+                        var raisePropertyChangedMethodReference = raisePropertyChangedMethod.MakeGenericMethod(genericTargetType);
+
+                        foreach (var method in setMethodsForGetInstructions)
+                        {
+                            method.Body.Emit(il =>
+                            {
+                                var last = method.Body.Instructions.Last(i => i.OpCode == OpCodes.Ret && (i.Next == null || method.Body.Instructions.Last() == i));
```

Comment:
I could also restrict so that it only works with dependent auto properties with the `[Reactive]` attribute to remove some of the edge cases, risk of it not behaving as expected with other weavers then though..?